### PR TITLE
Add test code for chrono and refactoring

### PR DESCRIPTION
**Pull request by pott-cho**
Close: #719
- Added test cases for Chrono-based logic, covering:
  - Date boundary conditions
  - Formatting variations
  - Empty-string handling
- Consolidated scattered `DateTime<Utc> → i64` conversion logic into a common function to enable proper testing
  - Added `parse_giganto_timestamp_ns`, `parse_sysmon_timestamp_ns`, and `parse_zeek_timestamp_ns`
  - Updated security_log `parse_*_timestamp` functions to the unified `parse_*_timestamp_ns` variants

### CODE REVIEW

**Review by sophie-cluml — COMMENTED** [review]
This change should be merged after the REproduce 0.26.0 release.

File: src/netflow.rs [hunk 1]
```diff
@@ -129,3 +121,25 @@ impl ParseNetflowDatasets for Netflow9 {
         Ok(events)
     }
 }
+
+fn netflow_timestamp(unix_secs: i64, nanos: u32) -> i64 {
```
Comment (sophie-cluml) [inline src/netflow.rs]: Could you change the `unix_secs` to u32 please? The rationale behind this request is that I think the conversion from u32 to i64 is to satisfy the type of chrono.

Comment (pott-cho) [reply src/netflow.rs]: I’ve updated it.
